### PR TITLE
telemetry: Enable data fields that were disabled

### PR DESCRIPTION
**Pull request by Pothulapati**
## Description

https://github.com/gitpod-io/gitpod/pull/8617 disabled the telemetry
fields as those database queries were causing heavy CPU performance.

https://github.com/gitpod-io/gitpod/pull/8858 fixed that issue by
adding indexes for those queries.

This PR also adds tracing for the telemetry data provider so that we
can also count how long they take usually, and how frequently these
are called in prod.

Signed-off-by: Tarun Pothulapati <tarun@gitpod.io>

## Related Issue(s)

Fixes #

## Release Notes

```release-note
[telemetry] enable data fields that were previously disabled
```

### CODE REVIEW

**Comment by roboquat** [comment]
@Pothulapati: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by kylos101** [comment]
/werft run with-helm=true with-observability=true with-clean-slate-deployment=true

**Comment by Pothulapati** [comment]
/werft run with-helm=true with-observability=true with-clean-slate-deployment=true

File: components/server/src/installation-admin/installation-admin-controller.ts [hunk 1]
```diff
@@ -17,7 +18,11 @@ export class InstallationAdminController {
         const app = express();
 
         app.get("/data", async (req: express.Request, res: express.Response) => {
+            const spanCtx =
+                opentracing.globalTracer().extract(opentracing.FORMAT_HTTP_HEADERS, req.headers) || undefined;
+            const span = opentracing.globalTracer().startSpan("telemetryDataEndpoint", { childOf: spanCtx });
             res.status(200).json(await this.telemetryDataProvider.getTelemetryData());
```
Comment (geropl) [inline components/server/src/installation-admin/installation-admin-controller.ts:24]: The pattern here is to wrap the actual code in a `try { code } finally { span.finish(); }` to make sure we do not leak resources.

File: components/server/src/installation-admin/telemetry-data-provider.ts [hunk 2]
```diff
@@ -16,13 +17,15 @@ export class InstallationAdminTelemetryDataProvider {
...
             installationAdmin: await this.installationAdminDb.getData(),
-            totalUsers: 0, //await this.userDb.getUserCount(true),
-            totalWorkspaces: 0, //await this.workspaceDb.getWorkspaceCount(),
-            totalInstances: 0, //await this.workspaceDb.getInstanceCount(),
+            totalUsers: await this.userDb.getUserCount(true),
+            totalWorkspaces: await this.workspaceDb.getWorkspaceCount(),
+            totalInstances: await this.workspaceDb.getInstanceCount(),
         } as TelemetryData;
 
+        span.finish();
```
Comment (geropl) [inline components/server/src/installation-admin/telemetry-data-provider.ts:28]: same here.

**Review by geropl — CHANGES_REQUESTED** [review]
THx for the PR!

Could you add the try...finally? Happy to quickly approve afterwards. :+1:

**Review by geropl — DISMISSED** [review]
Thanks!

**Comment by Pothulapati** [comment]
/werft run with-helm=true with-observability=true with-clean-slate-deployment=true

**Review by easyCZ — DISMISSED** [review]
LGTM

**Review by MrSimonEmms — DISMISSED** [review]

**Review by MrSimonEmms — APPROVED** [review]

**Review by easyCZ — APPROVED** [review]

**Pull request merged**